nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-ui/src/main/webapp/js/nf/canvas/nf-actions.js

```diff
@@ -845,6 +845,116 @@
             }
         },
 
+        /**
+         * Enable all controller services in the specified ProcessGroup.
+         *
+         * @argument {selection} selection      The selection
+         * @argument {cb} callback              The function to call when request is processed
+         */
+         enableAllControllerServices: function (selection,cb) {
+            if (selection.empty()) {
+                // build the entity
+                var entity = {
+                    'id': nfCanvasUtils.getGroupId(),
+                    'state': 'ENABLED'
+                };
+                updateResource(config.urls.api + '/flow/process-groups/' + encodeURIComponent(nfCanvasUtils.getGroupId()) + '/controller-services', entity).done(updateProcessGroup);
+            } else {
+                var componentsToStart = selection.filter(function (d) {
+                    return nfCanvasUtils.isProcessGroup(d3.select(this));
+                });
+
+                // ensure there are some component to start
+                if (!componentsToStart.empty()) {
+                    var startRequests = [];
+
+                    // start each selected component
+                    componentsToStart.each(function (d) {
+                        var selected = d3.select(this);
+
+                        // prepare the request
+                        var uri, entity;
+                        uri = config.urls.api + '/flow/process-groups/' + encodeURIComponent(d.id) + '/controller-services';
+                        entity = {
+                            'id': d.id,
+                            'state': 'ENABLED'
+                        };
+
+                        startRequests.push(updateResource(uri, entity).done(function (response) {
+                            nfCanvasUtils.getComponentByType('ProcessGroup').reload(d.id);
```

Comment:
In the context of starting a Processor, this reload call was necessary to update the Process Group node on the canvas. There currently are no Controller Service statuses rendered on the Process Group node. I don't believe this call is necessary here.

Reply:
Actually it updates the processors status after enable/disable de controllers. I think it's better to keep this.